master/internal/trial.go

```diff
@@ -278,7 +278,7 @@ func (t *trial) Receive(ctx *actor.Context) error {
 		}
 		ctx.Log().Info("trial stopped successfully")
 		endState := model.CompletedState
-		if t.experimentState == model.StoppingCanceledState {
+		if t.experimentState == model.StoppingCanceledState || t.killed {
```

Comment:
maybe drop a `// TODO: add a killed state` because it looks like it should be fairly easy to do.